npm_and_yarn/spec/dependabot/npm_and_yarn/file_updater_spec.rb

```diff
@@ -1487,6 +1487,46 @@
       end
     end
 
+    context "when 'latest' is specified" do
+      let(:manifest_fixture_name) { "latest_package_requirement.json" }
+      let(:npm_lock_fixture_name) { "latest_package_requirement.json" }
+      let(:yarn_lock_fixture_name) { "latest_package_requirement.lock" }
+
+      let(:version) { "3.0.2" }
+      let(:dependency_name) { "extend" }
+      let(:previous_version) { "2.0.1" }
+      let(:requirements) do
+        [{
+          file: "package.json",
+          requirement: "^3.0.2",
+          groups: ["dependencies"],
+          source: nil
+        }]
+      end
+      let(:previous_requirements) do
+        [{
+          file: "package.json",
+          requirement: "^2.0.1",
+          groups: ["dependencies"],
+          source: nil
+        }]
+      end
+
```

Comment:
As we where discussing on the call I think we should separate these tests for yarn and npm to make sure they are not messing with each other, here's an example where we do this: https://github.com/dependabot/dependabot-core/blob/master/npm_and_yarn/spec/dependabot/npm_and_yarn/file_updater_spec.rb#L987-L1009

So could then split up this `if` to check each package manager separately. Wanna give that a go?

Reply:
Agreed.  I'll work on that.

Reply:
I split up the tests, but when I run them, I continue to get a failure saying that the collection contains both the npm and yarn files.

Reply:
Ah yeah to run a spec for yarn only you'll need to set the `files` array to only the manifest and the yarn lock: `let(:files) { [package_json, yarn_lock] }` also you'll need the same for npm: `let(:files) { [package_json, package_lock] }`. It's set to both yarn and npm by default: https://github.com/dependabot/dependabot-core/blob/master/npm_and_yarn/spec/dependabot/npm_and_yarn/file_updater_spec.rb#L21

Because of rspec nesting magic we can clean this up a bit and specify the manifest files once in the top context and then create one nested context for yarn and one for npm:

```ruby
context "when 'latest' is specified as version requirement" do
  let(:manifest_fixture_name) { "latest_package_requirement.json" }
  let(:npm_lock_fixture_name) { "latest_package_requirement.json" }
  let(:yarn_lock_fixture_name) { "latest_package_requirement.lock" }

  let(:dependency_name) { "extend" }
  let(:version) { "3.0.2" }
  let(:previous_version) { "2.0.1" }
  let(:requirements) do
    [{
      file: "package.json",
      requirement: "^3.0.2",
      groups: ["dependencies"],
      source: nil
    }]
  end
  let(:previous_requirements) do
    [{
      file: "package.json",
      requirement: "^2.0.1",
      groups: ["dependencies"],
      source: nil
    }]
  end

  context "with a npm lockfile" do
    let(:files) { [package_json, package_lock] }

    it "only updates extend and locks etag" do
      expect(updated_files.map(&:name)).
        to match_array(%w(package.json package-lock.json))

      expect(updated_npm_lock.content).
        to include("extend/-/extend-3.0.2.tgz")
      expect(updated_npm_lock.content).
        to include("etag/-/etag-1.7.0.tgz")
    end
  end

  context "with a yarn lockfile" do
    let(:files) { [package_json, yarn_lock] }

    it "only updates extend and locks etag" do
      expect(updated_files.map(&:name)).
        to match_array(%w(package.json yarn.lock))

      expect(updated_yarn_lock.content).
        to include("extend@^3.0.2:\n  version \"3.0.2\"")
      expect(updated_yarn_lock.content).
        to include("etag@latest:\n  version \"1.7.0\"")
    end
  end
end
```

Reply:
Cool.  I definitely felt like I was missing something there.  That fixed the broken tests.

Also, nice idea on cleaning up the code.